modules/vaos/spec/services/v2/appointment_service_spec.rb

```diff
@@ -800,4 +802,150 @@
       end
     end
   end
+
+  describe '#extract_station_and_ien' do
+    it 'returns nil if the appointment does not have any identifiers' do
+      appointment = {}
+
+      expect(subject.send(:extract_station_and_ien, appointment)).to be_nil
+    end
+
+    it 'returns nil if the identifier with the system VistADefinedTerms/409_84 is not found' do
+      appointment = { identifier: [{ system: 'some_other_system', value: 'some_value' }] }
+
+      expect(subject.send(:extract_station_and_ien, appointment)).to be_nil
+    end
+
+    it 'returns the station id and ien if the identifier with the system VistADefinedTerms/409_84 is found' do
+      appointment = { identifier: [{ system: '/Terminology/VistADefinedTerms/409_84', value: '983:12345678' }] }
+      expected_result = %w[983 12345678]
+
+      expect(subject.send(:extract_station_and_ien, appointment)).to eq(expected_result)
+    end
+  end
+
+  describe '#avs_applicable?' do
+    before { travel_to(DateTime.parse('2023-09-26T10:00:00-07:00')) }
+    after { travel_back }
+
+    let(:past_appointment) { { status: 'booked', start: '2023-09-25T10:00:00-07:00' } }
+    let(:future_appointment) { { status: 'booked', start: '2023-09-27T11:00:00-07:00' } }
+    let(:unbooked_appointment) { { status: 'pending', start: '2023-09-25T10:00:00-07:00' } }
+
+    it 'returns true if the appointment is booked and is in the past' do
+      expect(subject.send(:avs_applicable?, past_appointment)).to be true
+    end
+
+    it 'returns false if the appointment is not booked' do
+      expect(subject.send(:avs_applicable?, unbooked_appointment)).to be false
+    end
+
+    it 'returns false on a booked future appointment' do
+      expect(subject.send(:avs_applicable?, future_appointment)).to be false
+    end
+  end
+
+  describe '#normalize_icn' do
+    context 'when icn is nil' do
+      it 'returns nil' do
+        expect(subject.send(:normalize_icn, nil)).to be_nil
+      end
+    end
+
+    context 'when icn is an empty string' do
+      it 'returns an empty string' do
+        expect(subject.send(:normalize_icn, '')).to eq('')
+      end
+    end
+
+    context 'when icn does not end with "V" followed by six digits' do
+      icn = '123456AA789012'
+
+      it 'returns the same icn' do
+        expect(subject.send(:normalize_icn, icn)).to eq(icn)
+      end
+    end
+
+    context 'when icn ends with "V" followed by six digits' do
+      icn = '1234567890V654321'
+
+      it 'removes trailing "V" followed by six digits' do
+        expect(subject.send(:normalize_icn, icn)).to eq('1234567890')
+      end
+    end
+  end
+
+  describe '#icns_match?' do
```

Comment:
I don't think you necessarily need to add the test case, but I believe these would also match if the first part were the same but the `Vxxxxxx` portion differed.

Reply:
I see. That makes sense.